subsys/mgmt/mcumgr/lib/cmd/shell_mgmt/src/shell_mgmt.c

```diff
@@ -10,16 +10,37 @@
 #include "mgmt/mgmt.h"
 #include "cborattr/cborattr.h"
 #include "shell_mgmt/shell_mgmt.h"
-#include "shell_mgmt/shell_mgmt_impl.h"
 #include "shell_mgmt/shell_mgmt_config.h"
+#include <shell/shell_dummy.h>
+#include <shell/shell_dummy.h>
+
+static int
+shell_exec(const char *line)
+{
+	const struct shell *shell = shell_backend_dummy_get_ptr();
+
+	shell_backend_dummy_clear_output(shell);
+	return shell_execute_cmd(shell, line);
+}
+
+const char *
+shell_get_output()
+{
+	size_t len;
+
+	return shell_backend_dummy_get_output(
+		shell_backend_dummy_get_ptr(),
+		&len
+	);
+}
 
 /**
  * Command handler: shell exec
  */
 static int
 shell_mgmt_exec(struct mgmt_ctxt *cb)
 {
-	static char line[SHELL_MGMT_MAX_LINE_LEN + 1] = {0};
+	static char line[SHELL_MGMT_MAX_LINE_LEN + 1];
```

Comment:
will this being static have issues if 2 SMP interfaces are active and both send a shell exec function at the same or similar time?

Reply:
Yes it will, you are right. I have failed to notice "static".
Removal of the static will increase required stack size. I should probably do that and update Kconfig for shell management to notify users on possible increased stack requirements.